source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -5906,16 +5906,16 @@ export abstract class StylableCardElementContainer extends CardElementContainer
     }
 
     protected getHasBackground(): boolean {
-        return this.getHasPadding(true);
+        return this.getHasPadding(GlobalSettings.addPaddingToContainerWithBackgroundImage);
```

Comment:
No - `CardElement.getHasBackground` MUST return false, always. This isn't the place where I suggested to make the change. Here is my original comment:

Plus the implementation of `StylableCardElementContainer.getDefaultPadding` is wonky:

```typescript
protected getDefaultPadding(): PaddingDefinition {
    let hasPadding = GlobalSettings.addPaddingToContainerWithBackgroundImage ? 
        this.getHasBackground() : this.getHasPadding(false);
    ...
```

It should be
```typescript
protected getDefaultPadding(): PaddingDefinition {
    let hasPadding = this.getHasPadding(GlobalSettings.addPaddingToContainerWithBackgroundImage);
    ...
```

Reply:
Sorry, I misread the change here. You're making the change in `StylableCardElementContainer.getHasBackground`, and not `CardElement.getHasBackground`. While that's better, it's still not good. Padding is determined by checking if a container has a background. So we cannot determine if a container has a background by checking if it has padding.

Reply:
@dclaux Thanks David; your offline suggestion really helped.